R/AnnotationFilter.R

```diff
@@ -373,6 +376,20 @@ NULL
     paste0(class, if (length(class)) "Filter" else character(0))
 }
 
+.classToField <- function(class) {
+  class <- sub("Filter$", "", class)
+  class <- gsub("([[:upper:]])", "_\\1", class, perl=TRUE)
+  class <- sub("^_","", class)
+  class <- tolower(class)
+  class
+}
+
+.supportedFiltersAndFields <- function() {
+  filters <- supportedFilters()
+  df <- data.frame(filter=filters, field=.classToField(filters))
+  df
```

Comment:
One thing: the mapping of `GRangesFilter` to a field `g_ranges` will not work - I suggest to list an `NA` for `GRangesFilter`.